linuxkit/linuxkit
# add the ability to extend linuxkit images

closes #3405 

**- What I did**

Added a `from` variable to the yaml schema. This allows the user to specify a docker image (atm it's a tar image due to https://github.com/linuxkit/linuxkit/issues/3052) to extend upon.

This allows the user to "layer" images by adding additional packages, files, ect... similar to adding layers to a docker image.

**- How I did it**

Modified the build process to use an existing image instead of a kernel if given. This will copy the kernel and cmdline from the existing image and layer everything else on top.

There may be some bugs and unexpected behaviours but after some quick tests it seems like everything is still working as expected.

**- How to verify it**

Build an image i.e `examples/sshd.yml` then build another image that uses the first i.e `from-sshd.yml`. Then run the image and the additional ntpd service should be running.

**- Description for the changelog**

Add the ability to extend and layer on top of images.
